pkgs/development/libraries/qt-5/5.12/default.nix

```diff
@@ -235,6 +235,8 @@ let
         };
       } ../hooks/qmake-hook.sh;
 
+      inherit stdenv;
```

Comment:
I feel this should be in passthru, as the derivation itself doesn't use it.

Reply:
Hm, I thought we were already in a package set here, not a derivation  ... are you sure that `passthru` would help?

In fact, if I naively just push it into `passthru.stdenv`:

```
error: attribute 'stdenv' missing

       at /home/code/nixpkgs/pkgs/top-level/all-packages.nix:29040:29:

        29039|   pulseview = libsForQt514.callPackage ../applications/science/electronics/pulseview {
        29040|     boost = boost.override { inherit (libsForQt514) stdenv; };
             |                             ^
        29041|   };
```

Reply:
Oh, you're right, it's not inside a derivation. (What's causing the big rebuild then?) I'm not sure the best way forward. Just take the rebuild hit?

Reply:
Every derivation also re-exposes its `stdenv`, so another solution could be e.g. `libsForQt514.qtbase.stdenv`. Would be nice to investigate the cause of the rebuild, though, before deciding on an implementation.

Reply:
> Would be nice to investigate the cause of the rebuild, though, before deciding on an implementation.

I reviewed the `antimony` derivation. It looks like none of the transitive build inputs have changed. Also the closure (`nix-store -qR`) is identical (modulo the antimony drv itself).

What did change is the stdenv of the antimony drv, which is now the one from qt. This also makes sense: overwriting stdenv in the libsForQt514 scope will change what `libsForQt514.callPackage` passes.

So essentially, this change will cause packages built via `libsForQt.{callPackage,mkDerivation,...}` and take `stdenv` to be receive the same stdenv, that qt uses. Arguably, this may be the more correct default .. any opinions on this?

Reply:
Right, I should've figured that out. The change sounds sane, although there may be complications in individual cases. Definitely needs to be retargeted to `staging`.

Reply:
However, this still requires boost to be overridden explicitly and effectively does nothing above that for pulseview or sigrok. 

So I was actually leaning towards reducing this PR to just use qtbase.stdenv, and open up a separate issue for `stdenv` in the qt scopes, which may also entail consolidating overrides for `boost` and other ABI sensitive packages.

Reply:
Also a possibility. Or add the (properly overridden) `boost` to the `libsForQt*` scope.